File: app/src/main/java/com/lbynet/phokus/camera/AnalysisResult.java
package com.lbynet.phokus.camera;

import android.media.Image;

public class AnalysisResult {

    private static Image prev = null;
    public static boolean is_paused = false;


    public static synchronized void put(Image proxy) {

        if(is_paused) return;

        prev = proxy;
    }

    public static synchronized Image get() {
        is_paused = true;
        return prev;
    }

    public static synchronized void unlock() {
        is_paused = false;

    }


}

File: app/src/main/java/com/lbynet/phokus/camera/AnalysisResult.kt
package com.lbynet.phokus.camera

import android.media.Image
import kotlin.jvm.Synchronized
import com.lbynet.phokus.camera.AnalysisResult

object AnalysisResult {

    private var prev: Image? = null
    var is_paused = false


    @Synchronized
    fun put(proxy: Image?) {
        if (is_paused) return
        prev = proxy
    }

    @Synchronized
    fun get(): Image? {
        is_paused = true
        return prev
    }

    @Synchronized
    fun unlock() {
        is_paused = false
    }
}